java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/broker/singlethread/SingleThreadedMessageBroker.java

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * Single threaded implementation of the {@link MessageBroker}.
+ *
+ * <p>Note that the {@link ConcurrentMessageBroker} with a concurrency level of one could have been used
+ * but there would have been an extra thread checking to see whether the concurrency rate has changed (which it never will)
+ * and therefore is wasted cycles. This implementation therefore can be simpler and less prone to errors.
+ */
+@Slf4j
+public class SingleThreadedMessageBroker extends AbstractMessageBroker {
+    private final MessageRetriever messageRetriever;
+    private final MessageProcessor messageProcessor;
+    private final ExecutorService executorService;
+
+    public SingleThreadedMessageBroker(final MessageRetriever messageRetriever, final MessageProcessor messageProcessor) {
+        this(messageRetriever, messageProcessor, Executors.newFixedThreadPool(2));
+    }
+
+    @VisibleForTesting
+    SingleThreadedMessageBroker(final MessageRetriever messageRetriever,
+                                final MessageProcessor messageProcessor,
+                                final ExecutorService executorService) {
+        super(executorService);
+
+        this.messageRetriever = messageRetriever;
+        this.messageProcessor = messageProcessor;
+        this.executorService = executorService;
+    }
+
+    @Override
+    protected SingleThreadMessageController createController(final CompletableFuture<Object> controllerFinishedFuture) {
+        return new SingleThreadMessageController(messageRetriever, messageProcessor, executorService, controllerFinishedFuture);
+    }
+
+    /**
+     * This is the controller that will handle the retrieval of messages and the subsequent processing.
+     *
+     * <p>This will only be run on a single thread and any message that is processed will be started on a new thread. The reason for starting it on a new
+     * thread is because we don't want interrupts to cause the processing of the message to be stopped. If the thread is interrupted it will wait until the
+     * message is processed for stopping the thread.
+     */
+    @Slf4j
+    @RequiredArgsConstructor
+    static class SingleThreadMessageController  implements AbstractMessageBroker.Controller {
+        /**
+         * The amount of time that the thread should be slept if there was an error retrieving messages for processing.
+         */
+        private static final int BACKOFF_TIME_MS = 1000;
+
+        private final MessageRetriever messageRetriever;
+        private final MessageProcessor messageProcessor;
+        private final ExecutorService executorService;
+        private final CompletableFuture<Object> controllerFinishedFuture;
+
+        /**
+         * When shutting down this will determine whether any messages currently being processed should be interrupted.
+         */
+        private boolean interruptThreads;
+
+        @Override
+        public void run() {
+            log.debug("Starting message listener");
```

Comment:
Add an equivalent log message in the concurrent one as well

Reply:
Yep!